geopandas/io/tests/test_file.py

```diff
@@ -840,15 +854,65 @@ def test_read_file__where_filter(engine, naturalearth_lowres):
             )
 
 
-@PYOGRIO_MARK
-def test_read_file__columns(naturalearth_lowres):
-    # TODO: this is only support for pyogrio, but we could mimic it for fiona as well
-    gdf = geopandas.read_file(
-        naturalearth_lowres,
-        columns=["name", "pop_est"],
-        engine="pyogrio",
-    )
-    assert gdf.columns.tolist() == ["name", "pop_est", "geometry"]
+def test_read_file__columns(engine, naturalearth_lowres):
+    if engine == "fiona" and not FIONA_GE_19:
```

Comment:
It is likely just my personal preference, but I would suggest skipping these new tests if Fiona < 1.9, and put a test for the exception handling for Fiona < 1.9 into its own test case (since behavior would be same for empty vs non-empty `columns`).